librepo/downloader.c

```diff
@@ -2349,8 +2357,11 @@ check_transfer_statuses(LrDownload *dd, GError **err)
                     target->tried_mirrors = g_slist_remove(target->tried_mirrors, target->mirror);
                     num_of_tried_mirrors = g_slist_length(target->tried_mirrors);
                 }
+                // complete_url_in_path and target->baseurl doesn't have an alternatives like using
+                // mirrors, therefore they are handled differently and conditions for retry download
+                // also differes
```

Comment:
At the end, i'd phrase it:
> ... conditions for retry also differ

(the fact that it's a download retry is clear from the context)